lazy-pages/src/sys/unix.rs

```diff
@@ -120,21 +138,28 @@ where
     };
 
     let old_sigaction = signal::sigaction(signal, &sig_action).map_err(io::Error::from)?;
-    OLD_SIG_HANDLER.with(|sh| *sh.borrow_mut() = old_sigaction.handler());
+    let handler = old_sigaction.handler();
+    let _ = OLD_SIG_HANDLER
+        .set(handler)
```

Comment:
Won't we face problem old handler is not saved in multithreading mode?

> Returns Ok(()) if the cell was empty and Err(value) if it was full.

Reply:
Signal handler is always only one for the whole process - so nvm

Reply:
All is ok